MinMaxGameHashed.java

```diff
@@ -1,76 +1,90 @@
 import java.util.HashMap;
 
 class MinMaxGameHashed extends MinMaxGame {
-	private HashMap<Integer, int[]> dataBase = new HashMap<Integer, int[]>(1000000);
-	private int starterPlayer;
+	protected HashMap<Integer, int[]> dataBase = new HashMap<Integer, int[]>(1000000);
 
 	public MinMaxGameHashed(int height, int width, boolean playersTurn) {
 		super(height, width, playersTurn);
-		if (playersTurn)
-			starterPlayer = 1;
-		else
-			starterPlayer = 2;
+		dataBase.put(0, null);
 	}
-	@override
-	protected int[] getValue(int[][] table, int lastMoveY, int lastMoveX, 
-								int player, int depth) {
-		// Opponent makes move (lastMoveY, lastMoveX) leading to table. 
-		// RETURNS: 
-		//    First entry:  The value of the position for player.
-		//    Second entry: depth + The lenght of the resulting game assuming optimal play.
-		// 
-		// NOTE: opponent = 3 - player. The second coordinate is needed so that the 
-		// computer player wants to win as quickly as possible or survive as long 
-		// as it can.
+	@Override
+	protected ValueWithMove getBestMove() {
+		// Getting best move at a current state of the table.
+		// RETURNS: Value of position and best move realizing that value.
```

Comment:
Comment could be phrased differently - in particular `realizing that value`

Reply:
I think it's okay